presto-hive/src/main/java/com/facebook/presto/hive/ColumnarTextHiveRecordCursor.java

```diff
@@ -324,7 +324,7 @@ public long getLong(int fieldId)
 
         if (!types[fieldId].equals(BIGINT) && !types[fieldId].equals(INTEGER) && !types[fieldId].equals(DATE) && !types[fieldId].equals(TIMESTAMP)) {
             // we don't use Preconditions.checkArgument because it requires boxing fieldId, which affects inner loop performance
-            throw new IllegalArgumentException(String.format("Expected field to be %s, %s or %s , actual %s (field %s)", BIGINT, DATE, TIMESTAMP, types[fieldId], fieldId));
+            throw new IllegalArgumentException(String.format("Expected field to be %s, %s, %s or %s , actual %s (field %s)", INTEGER, BIGINT, DATE, TIMESTAMP, types[fieldId], fieldId));
```

Comment:
Yep. This is my first stacked PR and I forgot to re-order my later fixes. I can merge before pushing.